libraries/standard/http/include/http_client.h

```diff
@@ -423,19 +457,30 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
  * parameter pRequestBodyBuf over the transport. The response is received in
  * #HTTPResponse_t.
  *
+ * The application should close the connection with the server if any
+ * HTTP_SECURITY_ALERT_X errors are returned. Any error found in parsing is
+ * considered a malformed response and therefore a security alert.
+ *
  * TODO: Expand documentation.
  *
  * @param[in] pTransport Transport interface, see #HTTPTransportInterface_t for
  * more information.
  * @param[in] pRequestHeaders Request configuration containing the buffer of
  * headers to send.
- * @param[in] pRequestBodyBuf Request entity body.
+ * @param[in] pRequestBodyBuf Optional Request entity body. Set to NULL if there
+ * is no request body.
  * @param[in] reqBodyBufLen The length of the request entity in bytes.
  * @param[in] pResponse The response message and some notable response
  * parameters will be returned here on success.
  *
- * @return #HTTP_SUCCESS if successful, an error code otherwise.
- * TODO: Update for exact error codes returned.
+ * @return One of the following:
+ * - #HTTP_SUCCESS
```

Comment:
Can you add scenarios when these error codes will be returned just like other function docs?

Reply:
This is not in the v4 docs because people are expected to go to the error code and read about. You guys went above and extra.

Reply:
We raised the bar!

Reply:
I think it's duplicating information.